_posts/2024-06-27-WildFly-playing-with-generative-ai.adoc

```diff
@@ -0,0 +1,319 @@
+---
+layout: post
+title:  "Playing with Generative AI with WildFly"
+date:   2024-06-27
+tags: generative-ai rag wildfly
+author: ehugonne
+description: How to try the Generative AI feature pack with WildFly
+---
+
+This blog post provides information on how to use the AI Galleon Feature pack to write generative AI applications.
+Please note that this feature pack is currently a proof of concept and might not be ready for production.
+
+
+== What's in this feature pack ?
+
+This feature pack is a work in progress so we will only cover what is already available as of today.
+It relies on configuring and exposing https://github.com/langchain4j[LangChain4J] to you application.
+It provides a way to inject some basic elements to make creating a Retrieval-Augmented Generation (RAG) application easy.
+It helps by configuring and providing all the required elements for such an application:
+
+
+=== What is RAG ?
+
+As its name indicates, RAG is about retrieving relevant elements from pieces of information from your data and they enrich the prompt with them before sending the whole query (initial prompt + relevant data) to the LLM. 
+The goal is to provide sufficient context to the LLM so that it can use those pieces of information and don't hallucinate.
+
+=== The core elements of RAG
+
+As you can see on the schema, you need a few elements to provide a RAG functionality:
+ * some components to get relevant pieces of information; we will focus on using a 'semantic' search approach aka vector searchso we will need:
+    - an embedding model to create embeddings (aka semantic vectors) from the user query thus creating an embedding query.
```

Comment:
a.k.a.

Reply:
Is there any good explanation that can be linked to from 'embeddings' or 'semantic vectors'?